react/features/base/react/functions.native.js

```diff
@@ -0,0 +1,37 @@
+// @flow
+
+import { Dimensions } from 'react-native';
+import Platform from './Platform';
+
+const IPHONEX_HEIGHT = 812;
+const IPHONEX_WIDTH = 375;
+
+/**
+* Determines if the device is an iPad or not.
+*
+* @returns {boolean}
+*/
+export function isIPad() {
+    const { height, width } = Dimensions.get('window');
+
+    return Platform.OS === 'ios' && (
+        Math.max(height, width)
+            / Math.min(height, width)) < 1.6;
+}
+
+/**
+* Determines if it's an iPhone X or not.
+*
+* @returns {boolean}
+*/
+export function isIPhoneX() {
+    const { height, width } = Dimensions.get('window');
```

Comment:
I don't like `isIPad` and `isIPhoneX`. We're in freaking react-native which allows us to write native source code to query iOS what the device is and we're writing some obscure source code involving the app's window dimensions and aspect rations?! Why are we OK with it, @saghul?

Reply:
Ok, so two things to explain here:

- The point of using an implementation like this is to minimise the blank area around screens while properly handle the iPhoneX top notch and iPhone's status bar (that behaves differently on iPad). I could have used React's new SafeArea component, but with screens that have a header (or similar, native looking screens) it renders a bad looking empty area around, that this implementation avoids.
- In another discussion with Saul we talked about introducing a new native module that would do similar things by getting these info from the device but it turned out that it doesn't solve all our problems so we decided not to introduce a new native module if it only gives us half of the solution.

Reply:
We made an attempt to use SafeArea as is, but IMHO it needs to be a larger change which other components such as WelcomePage and Conference also use, so I conceded going this route since the broken code was already in.